addons/TreasurePool/TreasurePool.lua

```diff
@@ -0,0 +1,160 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.]]
+
+_addon.name = 'TreasurePool'
+_addon.author = 'Kenshi'
+_addon.version = '1.0'
+
+require('luau')
+texts = require('texts')
+packets = require('packets')
+
+-- Config
+
+defaults = {}
+defaults.display = {}
+defaults.display.pos = {}
+defaults.display.pos.x = 0
+defaults.display.pos.y = 0
+defaults.display.bg = {}
+defaults.display.bg.red = 0
+defaults.display.bg.green = 0
+defaults.display.bg.blue = 0
+defaults.display.bg.alpha = 102
+defaults.display.bg.visible = true
+defaults.display.text = {}
+defaults.display.text.font = 'Consolas'
+defaults.display.text.red = 255
+defaults.display.text.green = 255
+defaults.display.text.blue = 255
+defaults.display.text.alpha = 255
+defaults.display.text.size = 12
+
+settings = config.load(defaults)
+settings:save()
+
+treasure_text = texts.new(settings.display, settings)
+
+treasure_text:appendline('Treasure Pool:')
+for i = 0, 9 do
+    treasure_text:appendline(i .. ': ${index' .. i .. '|-}${lotting' .. i .. '}')
+end
+
+goals = {}
+
+lotter = {}
+
+lot = {}
+
+windower.register_event('incoming chunk', function(id, data)
+    if id == 0x0D2 then
+    local packet = packets.parse('incoming', data)
+        -- Ignore gil drop
+        if packet.Item == 0xFFFF then
+            return
+        end
+    local time_check = packet.Timestamp + 300
+    local diff = os.difftime(time_check, os.time())
+        if diff <= 300 then
+            goals[packet.Index] = packet.Timestamp + 300
+            lotter[packet.Index] = ' '
+        else
+            goals[packet.Index] = os.time() + 300
+            lotter[packet.Index] = ' '
+        end
+    end
+    if id == 0x0D3 then
+    local lotpacket = packets.parse('incoming', data)
+        -- Ignore drop to a player or floored
+        if lotpacket.Drop ~= 0 then
+            return
+        else    
+            lotter[lotpacket.Index] = lotpacket['Highest Lotter Name']
+            lot[lotpacket.Index] = lotpacket['Highest Lot']
+        end
+    end
+    -- Check to hide text box if zoning with treasure up
+    if id == 0xB then
+        zoning_bool = true
+    elseif id == 0xA and zoning_bool then
+        zoning_bool = false
+    end
+end)
+
+windower.register_event('prerender', function()
+    local treasure = T(windower.ffxi.get_items().treasure)
+    local remove = S{}
+    local info = {}
+    if zoning_bool or treasure:empty() then
+        remove:add('index0') remove:add('index1') remove:add('index2') remove:add('index3') remove:add('index4') remove:add('index5') remove:add('index6') remove:add('index7') remove:add('index8') remove:add('index9')
+        remove:add('lotting0') remove:add('lotting1') remove:add('lotting2') remove:add('lotting3') remove:add('lotting4') remove:add('lotting5') remove:add('lotting6') remove:add('lotting7') remove:add('lotting8') remove:add('lotting9')
```

Comment:
Indeed those lines are redundant and I erased them, the code also changed a little to try to avoid some weird arithmetic on nil value error. I'll update it in a min.